devtools/src/tauri_plugin.rs

```diff
@@ -1,19 +1,60 @@
 use crate::aggregator::Aggregator;
 use crate::server::Server;
 use crate::Command;
+use std::collections::HashMap;
 use std::net::SocketAddr;
 use std::thread;
 use std::time::Duration;
 use tauri::Runtime;
 use tokio::sync::mpsc;
 
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord)]
+struct Key {
+    level: u16,
+    file: Option<String>,
+    line: Option<u32>,
+}
+
+#[allow(clippy::needless_pass_by_value)]
+#[tauri::command]
+fn log(
+    level: u16,
+    message: &str,
+    location: Option<&str>,
+    file: Option<&str>,
+    line: Option<u32>,
+    _key_values: Option<HashMap<String, String>>,
+) {
+    match level {
+        1 => {
+            tracing::trace!(target: "log", message, log.target = "webview", log.module_path = location, log.file = file, log.line = line);
+        }
+        2 => {
+            tracing::debug!(target: "log", message, log.target = "webview", log.module_path = location, log.file = file, log.line = line);
+        }
+        3 => {
+            tracing::info!(target: "log", message, log.target = "webview", log.module_path = location, log.file = file, log.line = line);
+        }
+        4 => {
+            tracing::warn!(target: "log", message, log.target = "webview", log.module_path = location, log.file = file, log.line = line);
+        }
+        5 => {
+            tracing::error!(target: "log", message, log.target = "webview", log.module_path = location, log.file = file, log.line = line);
+        }
+        _ => {}
+    }
+}
+
 pub(crate) fn init<R: Runtime>(
     addr: SocketAddr,
     publish_interval: Duration,
     aggregator: Aggregator,
     cmd_tx: mpsc::Sender<Command>,
 ) -> tauri::plugin::TauriPlugin<R> {
-    tauri::plugin::Builder::new("probe")
+    // we pretend to be the log plugin so we can intercept the commands
+    // this plugin is incompatible with the log plugin anyway
+    tauri::plugin::Builder::new("log")
```

Comment:
I don't really like this.. I feel if someone is using devtools, they shouldn't need the log plugin. We should document that instead (e.g. use the log plugin on production, devtools on development).

Reply:
Yes that is the idea, but as stated in the issue #150 devtools misses out the one feature that makes `tauri-plugin-log` great: the ability to emit logs from both Rust *and* JS to the same place.

Reply:
what do you think about making the log plugin not crash on init? so you can keep using its APIs

Reply:
yeah that could work actually

Reply:
gonna be difficult though, bc we have to init the log plugin in the `setup` callback not the builder function

Reply:
why does it matter? as long as we tell people to enable devtools first and both register at the same place..

Reply:
you mean like this?
```rust
tauri::Builder::default()
    .plugin(devtools::init())
    .plugin(tauri_plugin_log::init())
```

Reply:
yeah (plus where we actually register the log plugin)